bitcoin/tx.c

```diff
@@ -164,15 +165,30 @@ int bitcoin_tx_add_input(struct bitcoin_tx *tx, const struct bitcoin_txid *txid,
 	wally_err = wally_tx_input_init_alloc(txid->shad.sha.u.u8,
 					      sizeof(struct bitcoin_txid),
 					      outnum, sequence,
-					      script, tal_bytelen(script),
+					      scriptSig, tal_bytelen(scriptSig),
 					      NULL /* Empty witness stack */,
 					      &input);
 	assert(wally_err == WALLY_OK);
 	input->features = chainparams->is_elements ? WALLY_TX_IS_ELEMENTS : 0;
 	wally_tx_add_input(tx->wtx, input);
 	psbt_add_input(tx->psbt, input, i);
 
-
+	if (input_wscript)
+		/* Add the prev output's data into the PSBT struct */
+		psbt_input_set_prev_utxo_wscript(tx->psbt, i, input_wscript, amount);
+	else if (scriptPubkey) {
```

Comment:
```suggestion
	if (input_wscript) {
		/* Add the prev output's data into the PSBT struct */
		psbt_input_set_prev_utxo_wscript(tx->psbt, i, input_wscript, amount);
	} else if (scriptPubkey) {
```

Nit: just makes it clearer that the comment is not what's guarded by the `if`-statement :-)